core/controllers/base_test.py

```diff
@@ -2171,3 +2171,65 @@ def test_url_path_arg_normalization_is_successful(self):
             self.get_json(
                 '/mock_normalization/%s/%s' % (int_string, list_string),
                 expected_status_int=200)
+
+
+class RaiseErrorOnGetTest(test_utils.GenericTestBase):
+    """This test class is to ensure handlers with schema raises error
+    when they use self.request or self.payload."""
+
+    class MockHandlerWithSchema(base.BaseHandler):
+        """Mock handler with schema."""
+        URL_PATH_ARGS_SCHEMAS = {}
+        HANDLER_ARGS_SCHEMAS = {
+            'POST': {
+                'mock_int': {
+                    'schema': {
+                        'type': 'int'
+                    }
+                }
+            }
+        }
+
+        def post(self):
+            self.payload.get('mock_int')
+            return self.render_json({})
+
+    class MockHandlerWithoutSchema(base.BaseHandler):
+        """Mock handler without schema."""
+
+        def post(self):
+            self.payload.get('mock_int')
+            return self.render_json({})
+
+    def setUp(self):
+        super().setUp()
+        user_id = user_services.get_user_id_from_username('learneruser')
+        self.csrf_token = base.CsrfTokenManager.create_csrf_token(user_id)
+        self.payload = {'mock_int': 1}
+        self.testapp = webtest.TestApp(webapp2.WSGIApplication([
+            webapp2.Route('/mock_with_schema', self.MockHandlerWithSchema),
+            webapp2.Route(
+                '/mock_without_schema', self.MockHandlerWithoutSchema),
+        ], debug=feconf.DEBUG))
+
+    def test_raise_error_on_get(self):
+        err_message = 'err_message'
+        error = base.RaiseErrorOnGet(err_message)
```

Comment:
I'm not sure it's accurate to call this an "error". Perhaps object_which_raises_error_on_get or similar.

Reply:
Done

Reply:
I don't think this is done -- I'm referring to the local variable `error` on line 2193.

Reply:
Oh my bad. I changed the test name instead. Fixed it now. :)